### PR TITLE
Feature/#153 readd typescript to generator

### PR DESCRIPTION
**Pull request by janwidmer**
## Purpose of this pull request? 

* Enhancement

The pull request refers to the:

* generator (generator-nitro)

## What changes did you make?

Add a new (old) generator option to choose the JS Compiler to be either Typescript (Default) or Javascript

## Is there anything you'd like reviewers to focus on?

* Check if the jasmine tests pass, as I was not able to run them
* The tsconfig.json always seems to be created also for compiler JS, according to the generator code, it should not..

### CODE REVIEW

**Comment by ernscht** [comment]
Great, thanks 
Will have a look after my vacation.

File: packages/generator-nitro/generators/app/templates/package.json [hunk 1]
```diff
@@ -27,10 +27,10 @@
...
     "lint:css": "stylelint src/**/*.*ss --allow-empty-input",
     "lint:data": "nitro-app-validate-pattern-data",
-    "lint:html": "gulp lint-html",
-    "lint:js": "eslint ./src --ext .js",
-    "lint:license": "license-checker --production --summary --exclude \"Apache-2.0, BSD, ISC, LGPL, MIT, MPL\" --failOn \"AGPL; EPL; GPL\"",
-    "XXXlint:ts": "eslint ./src --ext .ts,.tsx --config .eslintrc-typescript.js",
+    "lint:html": "gulp lint-html", <% if (options.compiler === 'js') { %>
+    "lint:js": "eslint ./src --ext .js",<% } %>
+    "lint:license": "license-checker --production --summary --exclude \"Apache-2.0, BSD, ISC, LGPL, MIT, MPL\" --failOn \"AGPL; EPL; GPL\"",<% if (options.compiler === 'ts') { %>
+    "lint:ts": "eslint ./src --ext .ts,.tsx --config .eslintrc.js",<% } %>
```
Comment (ernscht) [inline packages/generator-nitro/generators/app/templates/package.json]: we don't need `--config`

File: packages/generator-nitro/generators/app/templates/package.json [hunk 2]
```diff
@@ -148,17 +154,17 @@
...
       "npm run lint:data",
       "git add"
-    ],
+    ],<% if (options.compiler === 'js') { %>
     "src/**/*.{js,jsx}": [
       "prettier --write",
       "eslint",
       "git add"
-    ],
+    ],<% } %><% if (options.compiler === 'ts') { %>
```
Comment (ernscht) [inline packages/generator-nitro/generators/app/templates/package.json]: would handle it in `"src/**/*.{js,jsx}": [`
and only switch the extensions (config not needed)

File: packages/generator-nitro/generators/app/templates/package.json (hunk 1 above)
Comment (janwidmer) [reply packages/generator-nitro/generators/app/templates/package.json]: removed.

File: packages/generator-nitro/generators/app/templates/package.json (hunk 2 above)
Comment (janwidmer) [reply packages/generator-nitro/generators/app/templates/package.json]: done

File: packages/generator-nitro/generators/app/index.js [hunk 3]
```diff
@@ -345,6 +379,28 @@ module.exports = class extends Generator {
 				}
 			}
 
+			// check if it's a src file
+			if (fileWithoutExt.indexOf('src/') !== -1) {
```
[the remark this answers is not in the document]
Comment (janwidmer) [reply packages/generator-nitro/generators/app/index.js]: enhanced if statement.

**Pull request merged**